### PR TITLE
refactor(MembershipsSdkAdapter): sort the members in a meeting

### PR DESCRIPTION
**Pull request by CernaianuMihai**
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-223552

### CODE REVIEW

File: src/MembershipsSDKAdapter.js [hunk 1]
```diff
@@ -39,8 +56,8 @@ const MAX_MEMBERSHIPS = 1000;
  * @returns {Array} List of active users in a meeting
  */
 function getMembers(members) {
-  return members ? Object.values(members)
-    .filter((member) => member.isUser)
+  return members ? sortMembers(Object.values(members)
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:59]: I would suggest this change, to make the data flow a little easier to follow:
```js
members = Object.values(members || {});
members = members.filter((member) => member.isUser);
members = sortMeetingMembers(members);
return members.map(...);
```

this way, the processing steps on the members are listed in chronological order.

File: src/MembershipsSDKAdapter.js [hunk 2]
```diff
@@ -31,6 +31,23 @@ const MAX_MEMBERSHIPS = 1000;
  * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/MembershipsAdapter.js#L6}
  */
 
+/**
+ *
+ * @param {object} members Members object from meeting, keyed by ID
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:36]: the input is a member array

File: src/MembershipsSDKAdapter.js [hunk 3]
```diff
@@ -31,6 +31,23 @@ const MAX_MEMBERSHIPS = 1000;
...
+ * @returns {Array} Sorted (by displayName) list of active users in a meeting
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:37]: just "name"

File: src/MembershipsSDKAdapter.js [hunk 4]
```diff
@@ -31,6 +31,23 @@ const MAX_MEMBERSHIPS = 1000;
...
+ */
+function sortMembers(members) {
+  return members.sort((member1, member2) => {
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:40]: one approach to having a single return:
```js
return members.sort((member1, member2) => (
  member1.isSelf ? -1 : // current user comes first
  member2.isSelf ? 1 :
  !member1.name ? 1 : // empty names come last
  !member2.name ? -1 :
  member1.name.localeCompare(member2.name)  // alphabetical order
);
```

File: src/mockSdk.js [hunk 5]
```diff
@@ -34,6 +36,7 @@ export const mockSDKMeeting = {
         },
         person2: {
           id: 'mutedPerson',
+          name: 'SDK',
```
Comment (cipak) [inline src/mockSdk.js:39]: let's use person names that contain firstnames and lastnames, to be closer to "reality". Maybe use the same names as here? https://github.com/webex/components/blob/master/src/data/people.js#L10

File: src/MembershipsSDKAdapter.js [hunk 6]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
  * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/MembershipsAdapter.js#L6}
  */
 
+/**
+ * Sort the members alphabeticaly but puts the current user first
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:35]: Sort the members alphabetically, with the current user first

File: src/MembershipsSDKAdapter.js (hunk 4 above)
Comment (cipak) [reply src/MembershipsSDKAdapter.js:41]: @lalli-flores there is an issue with this approach, the linter complains about nested ternary operators. Would it be ok to disable that check for this instance?

File: src/MembershipsSDKAdapter.js [hunk 7]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
...
+ *
+ * @param {Array} members List of members from meeting
+ * @returns {Array} Sorted list of users in a meeting
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:38]: maybe "Sorted list of members" (let's not change the term from member to user)

File: src/MembershipsSDKAdapter.js [hunk 8]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
...
+  });
+}
+
 /**
  * Gets the active members in a meeting
  *
  * @private
- * @param {object} members Members object from meeting, keyed by ID
- * @returns {Array} List of active users in a meeting
+ * @param {object} sdkMembers Members object from meeting
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:55]: keyed by ID is fine, please don't remove it

File: src/MembershipsSDKAdapter.js [hunk 9]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
...
+ * @returns {Array} List of users in a meeting
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:56]: you can use the adapter `Member` type here:
```js
* @returns {Member[]} List of meeting members
```

and please remove the "active" word from the function description on line 52, too.

File: src/MembershipsSDKAdapter.js [hunk 10]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
  * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/MembershipsAdapter.js#L6}
  */
 
+/**
+ * Sort the members alphabetically, with the current user first
+ *
+ * @param {Array} members List of members from meeting
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:37]: input: List of sdk meeting members
output: Sorted list of sdk meeting members

File: src/MembershipsSDKAdapter.js [hunk 11]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
...
+  });
+}
+
 /**
  * Gets the active members in a meeting
  *
  * @private
- * @param {object} members Members object from meeting, keyed by ID
- * @returns {Array} List of active users in a meeting
+ * @param {object} sdkMembers Members object from meeting, keyed by ID
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:55]: from sdk meeting

File: src/MembershipsSDKAdapter.js [hunk 12]
```diff
@@ -31,23 +31,43 @@ const MAX_MEMBERSHIPS = 1000;
...
+ * @returns {Member[]} List of meeting members
```
Comment (lalli-flores) [inline src/MembershipsSDKAdapter.js:56]: For consistency
```suggestion
 * @returns {Array.<Member>} List of meeting members
```

File: src/MembershipsSDKAdapter.js (hunk 4 above)
Comment (lalli-flores) [reply src/MembershipsSDKAdapter.js:41]: It should be ok to disable the linter for this one line but not at the module level

File: src/MembershipsSDKAdapter.js [hunk 13]
```diff
@@ -31,24 +31,44 @@ const MAX_MEMBERSHIPS = 1000;
...
+/**
+ * Sort the members alphabetically, with the current user first
+ *
+ * @param {Array} members List of sdk meeting members
+ * @returns {Array} Sorted list of sdk meeting members
+ */
+function sortMeetingMembers(members) {
+  return members.sort((member1, member2) => (
+  /* eslint-disable no-nested-ternary */
+    member1.isSelf ? -1// current user comes first
```
Comment (cipak) [inline src/MembershipsSDKAdapter.js:43]: please add a single space before the comment

Comment (cipak) [reply src/MembershipsSDKAdapter.js:43]: could you also disable the indentation rule that prevents the vertical aligning of these conditions?

File: src/MembershipsSDKAdapter.js [hunk 14]
```diff
@@ -31,24 +31,44 @@ const MAX_MEMBERSHIPS = 1000;
...
-      orgID: member.participant && member.participant.person && member.participant.person.orgId,
-      inMeeting: member.isInMeeting,
-      muted: member.isAudioMuted,
-      sharing: member.isContentSharing,
-      host: member.isHost,
-    })) : [];
+function getMembers(sdkMembers) {
+  let members = Object.values(sdkMembers || {});
+
+  members = members.filter((member) => member.isUser);
```
Comment (lalli-flores) [inline src/MembershipsSDKAdapter.js:61]: Just thinking of optimizing here, is there a way to do this check while sorting?

**Review by cipak — APPROVED** [review]

**Review by lalli-flores — APPROVED** [review]

File: src/MembershipsSDKAdapter.js (hunk 14 above)
Comment (lalli-flores) [reply src/MembershipsSDKAdapter.js:61]: @CernaianuMihai

**Pull request merged**